conan.cmake

```diff
@@ -197,19 +197,30 @@ endmacro()
 
 macro(conan_cmake_run)
     parse_arguments(${ARGV})
-    conan_cmake_generate_conanfile(${ARGV})
-    if(CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+
+    # TODO: Better detection
+    if(CONAN_EXPORTED OR CONAN_COMPILER)
```

Comment:
I am not sure if I'd use CONAN_COMPILER for detection, some users might be feeding it from the command line to cmake. CONAN_EXPORTED would do the task, wouldn't it? Is it necessary to set it to ``ON`` then?